flow/util/misc.py

```diff
@@ -251,3 +251,46 @@ def _cached_partial(func, *args, maxsize=None, **kwargs):
 
     """
     return lru_cache(maxsize=maxsize)(partial(func, *args, **kwargs))
+
+
+class bidict(dict):
+    r"""A bidirectional dictionary.
+
+    The attribute ``inverse`` contains the inverse mapping, where the inverse
+    values are :class:`set`\ s of keys with that value.
+
+    Both keys and values must be hashable.
+    A key is associated with one or more values.
+    A value is associated with exactly one key.
```

Comment:
I believe this is backwards. One value (e.g. aggregate store) can be associated with multiple keys (e.g. groups).

Reply:
Good catch. I wrote this docstring 3 times to fix other mistakes I'd made. 😆